File: Python/Symmetric Tree.py
#!/usr/bin/env python

class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None
        # @param root, a tree node
        # @return a boolean
def symmetric(self, root1, root2):
    if root1 == None and root2 == None:
        return True
    elif root1 == None or root2 == None:
        return False
    if root1.val != root2.val:
        return False
    return self.symmetric(root1.left, root2.right) and self.symmetric(root1.right, root2.left)

def isSymmetric(self, root):
    if root == None:
        return True
    elif root.left == None and root.right == None:
        return True
    elif root.left != None and root.right != None:
        return self.symmetric(root.left, root.right)
    else:
        return False

File: C++/Symmetric-Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    bool helper(TreeNode* root1, TreeNode* root2) {
        if (root1 == root2) return true;
        if (root1 == NULL || root2 == NULL) return false;
        if (root1->val == root2->val) {
            if (helper(root1->left, root2->right) && helper(root1->right, root2->left))
                return true;
        }
        return false;
    }
    bool isSymmetric(TreeNode* root) {
        if (root == NULL) return true;
        return helper(root->left, root->right);
    }
};